app/models/deterministic_assignment_creation.rb

```diff
@@ -14,13 +14,15 @@ def self.create!(params)
   end
 
   def save!
-    ArbitraryAssignmentCreation.create!(
-      visitor_id: visitor_id,
-      split_name: split_name,
-      variant: variant,
-      mixpanel_result: mixpanel_result,
-      context: context
-    )
+    unless split.feature_gate?
+      ArbitraryAssignmentCreation.create!(
```

Comment:
chatted offline, @jmileham pointed out that if product still wants to know how many people have seen the enabled experience (e.g. we've turned the feature flag on for 20% of people and want to gauge if there are any blocking bugs) we can query redshift's `customer_prod.split_assigned` table because segment will still track assignment events 💯

Reply:
excellent. for debugging purposes i think there is value in continuing to fire the `SplitAssigned` event to the analytics client that is configured by the client code.